# V6 Classification Manual Verification

Verify GPT-4o's classification of GPT-5 FOLIO baseline false negatives (27 cases).

In [21]:
import json
import pandas as pd
import re

# Load v6 classification results
df = pd.read_csv('../results/error_analysis/gpt-5_folio_baseline_v6.csv')

# Load FOLIO data
with open('../data/folio/original/folio-validation.json') as f:
    folio_data = json.load(f)

# Load baseline results
with open('../results/simplelean/gpt-5_folio_baseline/results.jsonl') as f:
    baseline = {r['case_idx']: r for r in [json.loads(l) for l in f]}

print(f'Loaded {len(df)} classifications')
print(f'\nCategory distribution:')
print(df['root_cause_category'].value_counts())

Loaded 27 classifications

Category distribution:
root_cause_category
AXIOM_FABRICATION    11
FAITHFUL              8
GOAL_MISMATCH         3
DATASET_BUG           2
REASONING_ERROR       1
VACUOUS_TRUTH         1
MISSING_PREMISE       1
Name: count, dtype: int64


In [22]:
def show_case(case_idx):
    """Display full details for a case."""
    row = df[df['case_idx'] == case_idx].iloc[0]
    folio = folio_data[case_idx]
    result = baseline[case_idx]
    
    print('=' * 70)
    print(f'CASE {case_idx}')
    print('=' * 70)
    print(f'Ground Truth: {row["ground_truth"]}')
    print(f'Prediction: {row["prediction"]}')
    print(f'V6 Category: {row["root_cause_category"]}')
    print(f'V6 Explanation: {row["error_description"]}')
    print(f'Problematic Element: {row["problematic_axiom"]}')
    print()
    print('--- PREMISES ---')
    print(folio['premises'])
    print()
    print('--- CONCLUSION ---')
    print(folio['conclusion'])
    print()
    print('--- LEAN CODE ---')
    print(result.get('lean_code', 'N/A'))

## AXIOM_FABRICATION Cases (11)

Model invented facts not in premises.

In [23]:
fab_cases = df[df['root_cause_category'] == 'AXIOM_FABRICATION']['case_idx'].tolist()
print(f'AXIOM_FABRICATION cases: {fab_cases}')

AXIOM_FABRICATION cases: [36, 29, 41, 46, 118, 70, 141, 77, 5, 92, 89]


In [24]:
# Case 70 - Known gaming case
show_case(70)

CASE 70
Ground Truth: Uncertain
Prediction: True
V6 Category: AXIOM_FABRICATION
V6 Explanation: Axiom A5_stock claims KO is a stock, but premises only say 'mature stock'
Problematic Element: A5_stock

--- PREMISES ---
All growth stocks are bought to earn profits from rapid price appreciation.
If the stock price is bought to earn profits from rapid price appreciation, then it is not suitable for a retirement fund.
Some stocks are growth stocks.
All mature stocks are suitable for a retirement fund.
KO is a mature stock.

--- CONCLUSION ---
KO is a stock.

--- LEAN CODE ---
axiom obj : Type
axiom KO : obj

axiom Stock : obj → Prop
axiom Growth : obj → Prop
axiom Mature : obj → Prop
axiom BoughtRapid : obj → Prop
axiom SuitableRetirement : obj → Prop

-- All growth stocks are bought to earn profits from rapid price appreciation.
axiom A1 : ∀ x : obj, Stock x ∧ Growth x → BoughtRapid x
-- If a stock is bought to earn profits from rapid price appreciation, then it is not suitable for a retir

**Case 70 Verdict:** CORRECT - Model added `axiom A5_stock : Stock KO` but premises only say "KO is a mature stock", not that KO is a stock.

In [25]:
# Case 89 - Questionable classification
show_case(89)

CASE 89
Ground Truth: Uncertain
Prediction: True
V6 Category: AXIOM_FABRICATION
V6 Explanation: The axiom K1 states that all books contain knowledge, which is not explicitly mentioned in the premises. The premises only state that books contain tons of knowledge, not all books.
Problematic Element: K1

--- PREMISES ---
Books contain tons of knowledge.
When a person reads a book, that person gains knowledge. 
If a person gains knowledge, they become smarter.
Harry read the book “Walden” by Henry Thoreau.

--- CONCLUSION ---
A smarter person has gained knowledge.

--- LEAN CODE ---
axiom obj : Type
axiom Person : obj → Prop
axiom Book : obj → Prop
axiom Reads : obj → obj → Prop
axiom GainsKnowledge : obj → Prop
axiom Smarter : obj → Prop
axiom ContainsKnowledge : obj → Prop

axiom Harry : obj
axiom Walden : obj

-- Books contain tons of knowledge.
axiom K1 : ∀ b : obj, Book b → ContainsKnowledge b
-- When a person reads a book, that person gains knowledge.
axiom R1 : ∀ p b : obj, Person p

**Case 89 Verdict:** WRONG CLASSIFICATION

V6 says AXIOM_FABRICATION because K1 "all books contain knowledge" vs premise "books contain tons of knowledge".

BUT the proof doesn't use K1! The proof chain is:
1. Harry read Walden (H3)
2. Walden is a book (H2), Harry is a person (H1)
3. Person reads book → gains knowledge (R1) → Harry gains knowledge
4. Gains knowledge → smarter (R2) → Harry is smarter
5. Therefore ∃p. Person p ∧ Smarter p ∧ GainsKnowledge p

**Should be: FAITHFUL** (model's reasoning is correct, GT may be wrong)

In [26]:
# Case 36
show_case(36)

CASE 36
Ground Truth: True
Prediction: Uncertain
V6 Category: AXIOM_FABRICATION
V6 Explanation: The axiom R2 introduces the concept of conductors leading orchestras, which is not explicitly stated in the premises.
Problematic Element: R2

--- PREMISES ---
Symphony No. 9 is a music piece.
Composers write music pieces.
Beethoven wrote Symphony No. 9.
Vienna Music Society premiered Symphony No. 9.
Vienna Music Society is an orchestra.
Beethoven leads the Vienna Music Society.
Orchestras are led by conductors.

--- CONCLUSION ---
Beethoven is a composer.

--- LEAN CODE ---
axiom obj : Type
axiom Beethoven : obj
axiom Symphony9 : obj
axiom ViennaMusicSociety : obj

axiom MusicPiece : obj → Prop
axiom Composer : obj → Prop
axiom Wrote : obj → obj → Prop
axiom Premiered : obj → obj → Prop
axiom Orchestra : obj → Prop
axiom Leads : obj → obj → Prop
axiom Conductor : obj → Prop

-- Symphony No. 9 is a music piece.
axiom T1 : MusicPiece Symphony9
-- Composers write music pieces. (If x is a compo

**Case 36 Verdict:** PARTIALLY CORRECT

V6 says AXIOM_FABRICATION for R2 "conductors leading orchestras".

Actual issues:
1. R2: Premise "Orchestras are led by conductors" → `Leads l o → Conductor l` is **wrong direction**
   - Premise says: Conductor → LeadsOrchestra  
   - Axiom says: LeadsOrchestra → Conductor
2. R1: Premise "Composers write music pieces" → `Composer x → MusicPiece y → Wrote x y` 
   - This says every composer wrote every piece!
   - Should be: `Wrote x y ∧ MusicPiece y → Composer x`

**Better classification: FORMALIZATION_ERROR** (wrong implication direction)

In [27]:
# Case 46
show_case(46)

CASE 46
Ground Truth: True
Prediction: Uncertain
V6 Category: AXIOM_FABRICATION
V6 Explanation: The axiom 'Student : Person → Prop' introduces the concept of a student, which is not explicitly mentioned in the premises.
Problematic Element: axiom Student : Person → Prop

--- PREMISES ---
There are four seasons in a year: Spring, Summer, Fall, and Winter.
All students who want to have a long vacation have summer as their favorite season.
Emma's favorite season is summer.
Mia's favorite season is not the same as Emma's. 
James wants to have a long vacation.

--- CONCLUSION ---
James's favorite season is summer.

--- LEAN CODE ---
axiom Person : Type
axiom Season : Type

-- Seasons
axiom Spring : Season
axiom Summer : Season
axiom Fall : Season
axiom Winter : Season

-- People
axiom Emma : Person
axiom Mia : Person
axiom James : Person

-- Predicates and functions
axiom FavoriteSeason : Person → Season
axiom Student : Person → Prop
axiom WantsLongVacation : Person → Prop

-- Rules and fac

**Case 46 Verdict:** NL AMBIGUITY

V6 says AXIOM_FABRICATION for `Student : Person → Prop`.

But premise says "All students who want to have a long vacation..." - so "Student" IS mentioned.

The real issue might be: Does "James wants a long vacation" mean James IS a student?
- Premise: "All students who want long vacation → summer favorite"
- This doesn't say wanting long vacation IMPLIES being student

**Should be: FORMALIZATION_ERROR or NL_AMBIGUITY** (not fabrication)

In [28]:
# Case 118
show_case(118)

CASE 118
Ground Truth: True
Prediction: Uncertain
V6 Category: AXIOM_FABRICATION
V6 Explanation: The axiom T8 states that Barutin is located in Bulgaria, which is not mentioned in the premises.
Problematic Element: T8

--- PREMISES ---
Barutin Cove is a cove named after the Bulgarian settlement of Barutin.
Barutin Cove is on the southwest coast of Snow Island.
Snow Island, Greenwich Island, and Deception Island are located in the South Shetland Islands.
Antarctica is located on the South Shetland Islands. 
If place A is located in place B and place B is located in place C, then place A is located in place C.

--- CONCLUSION ---
There is at least one cove in Antarctica named after a place in Bulgaria.

--- LEAN CODE ---
axiom obj : Type

-- Entities
axiom BarutinCove : obj
axiom Barutin : obj
axiom SnowIsland : obj
axiom GreenwichIsland : obj
axiom DeceptionIsland : obj
axiom SouthShetlandIslands : obj
axiom Antarctica : obj
axiom Bulgaria : obj

-- Predicates/Relations
axiom Cove : obj

**Case 118 Verdict:** NL AMBIGUITY / WORLD KNOWLEDGE

V6 says AXIOM_FABRICATION for T8 "Barutin is located in Bulgaria".

Premise says: "Barutin Cove is a cove named after the **Bulgarian settlement** of Barutin."

The model inferred that a "Bulgarian settlement" is in Bulgaria - this is world knowledge / reasonable inference.

**Should be: NL_AMBIGUITY or WORLD_KNOWLEDGE** (not pure fabrication)

In [29]:
# Case 29 - CONTRADICTORY V6 output
show_case(29)

CASE 29
Ground Truth: False
Prediction: Uncertain
V6 Category: AXIOM_FABRICATION
V6 Explanation: The Lean code does not contain any axiom that is not mentioned in the premises. All axioms are faithful to the premises provided.
Problematic Element: nan

--- PREMISES ---
All aliens are extraterrestrials.
If someone is from Mars, then they are an alien.
No extraterrestrials are human.
All highly intelligent beings from Earth are humans.
Marvin is a highly intelligent being.
Marvin is either from Earth and from Mars, or he is from neither.
If Marvin is not from Earth, then Marvin is an extraterrestrial.

--- CONCLUSION ---
If Marvin is not from Mars, then Marvin is a human.

--- LEAN CODE ---
axiom obj : Type
axiom Marvin : obj

axiom Alien : obj → Prop
axiom Extraterrestrial : obj → Prop
axiom Human : obj → Prop
axiom FromMars : obj → Prop
axiom FromEarth : obj → Prop
axiom HighlyIntelligent : obj → Prop

-- All aliens are extraterrestrials.
axiom A1 : ∀ x : obj, Alien x → Extraterrestria

**Case 29 Analysis**

**GT: False, Prediction: Uncertain (Conservative Error)**

**Premises:**
- All aliens are extraterrestrials.
- If someone is from Mars, then they are an alien.
- No extraterrestrials are human.
- All highly intelligent beings from Earth are humans.
- Marvin is a highly intelligent being.
- Marvin is either from Earth and from Mars, or he is from neither.
- If Marvin is not from Earth, then Marvin is an extraterrestrial.

**Conclusion:** If Marvin is not from Mars, then Marvin is a human.

**V6 Classification:** AXIOM_FABRICATION
**V6 Explanation:** "The Lean code does not contain any axiom that is not mentioned in the premises. All axioms are faithful to the premises provided."

**Verdict: V6 SELF-CONTRADICTION!**

The classification says AXIOM_FABRICATION but the explanation says "all axioms are faithful"! 

Looking at the Lean code, all axioms A1-A7 correctly match the premises. The model correctly shows both cases:
1. If Marvin is from both Earth and Mars → implication holds vacuously
2. If Marvin is from neither → implication is false (¬Human)

**Should be: FAITHFUL** - the formalization is correct, prediction is reasonable

In [30]:
# Case 92 - Suspicious classification
show_case(92)

CASE 92
Ground Truth: False
Prediction: Uncertain
V6 Category: AXIOM_FABRICATION
V6 Explanation: The axiom 'deposit_required' states that a deposit is required to rent an apartment in a managed building, but this is not explicitly mentioned in the premises.
Problematic Element: deposit_required

--- PREMISES ---
Pets are allowed in some managed buildings.
A deposit is required to rent an apartment in a managed building.
The security deposit can be either equal to the monthly rent at a managed building or more.
Fluffy is Tom's cat.
Cats are pets.
The Olive Garden is a managed building.
The monthly rent at the Olive Garden is $2000.
$2000 is more than $1500.
Tom will rent an apartment in a managed building if he is allowed to move in with Fluffy, and the security deposit is no more than $1500.
If a managed building allows pets, then people are allowed to move in with a pet. 

--- CONCLUSION ---
Tom will rent an apartment in The Olive Garden.

--- LEAN CODE ---
axiom Person : Type
axiom A

**Case 92 Analysis**

**GT: False, Prediction: Uncertain (Conservative Error)**

**Premises:**
- Pets are allowed in some managed buildings.
- **A deposit is required to rent an apartment in a managed building.**
- The security deposit can be either equal to the monthly rent at a managed building or more.
- Fluffy is Tom's cat.
- Cats are pets.
- The Olive Garden is a managed building.
- The monthly rent at The Olive Garden is $2000.
- Tom will rent an apartment in a managed building if and only if he is allowed to move in with Fluffy and the security deposit is no more than $1500.

**Conclusion:** Tom will rent an apartment in The Olive Garden.

**V6 Classification:** AXIOM_FABRICATION
**V6 Explanation:** "The axiom 'deposit_required' states that a deposit is required to rent an apartment in a managed building, but this is not explicitly mentioned in the premises."

**Verdict: V6 COMPLETELY WRONG!**

The premise **literally says**: "A deposit is required to rent an apartment in a managed building."

The axiom `deposit_required : ∀ b, Managed b → ∃ d, SecurityDeposit b d` is a direct translation!

**Should be: FAITHFUL** - V6 failed to read the premise correctly

In [31]:
# Case 77 - From bad story 368
show_case(77)

CASE 77
Ground Truth: False
Prediction: True
V6 Category: AXIOM_FABRICATION
V6 Explanation: The axiom H_need_imp introduces a condition about Hannah that is not present in the premises. Specifically, it states that if Hannah needs to earn money, then she is neither picky nor needs to earn money, which is not mentioned in the premises.
Problematic Element: H_need_imp

--- PREMISES ---
If people at Mary's school work in student jobs on campus, then they need to earn money to help pay for their college tuition.
If people at Mary's school order takeout frequently in college, then they work in student jobs on campus.
People at Mary's school order takeout frequently in college or enjoy the dining hall meals and recipes.
If people at Mary's school enjoy the dining hall meals and recipes, then they are not picky eaters.
If people at Mary's school enjoy the dining hall meals and recipes, then they spend a lot of their time eating and catching up with friends in the campus dining halls.
Hannah i

**Case 77 Analysis**

**GT: False, Prediction: True (Gaming)**

**Story 368** (same as Case 75 which V6 correctly classified as DATASET_BUG)

**Premises:**
- If people at Mary's school work in student jobs on campus, then they need to earn money to help pay for their college tuition.
- If people at Mary's school order takeout frequently in college, then they work in student jobs on campus.
- People at Mary's school order takeout frequently in college or enjoy eating in campus dining halls.
- If people at Mary's school enjoy eating in the campus dining halls, then they are not picky about food.
- If people at Mary's school are picky about food, they do not enjoy eating in the campus dining halls.
- Hannah is at Mary's school and she does not need to earn money to help pay for her college tuition. ← **CONTRADICTS premise chain!**
- Hannah is at Mary's school and if she is not picky about food, she spends a lot of her time eating and catching up with friends in the campus dining halls.

**Conclusion:** Hannah is at Mary's school and she either is not a picky eater or, if she is, then she spends a lot of her time eating and catching up with friends in the campus dining halls.

**V6 Classification:** AXIOM_FABRICATION (H_need_imp is fabricated)
**Case 75 (same story):** DATASET_BUG (correct!)

**Verdict: V6 INCONSISTENT!**

The premises contain a contradiction:
1. Everyone at Mary's school either orders takeout OR enjoys dining halls
2. Ordering takeout → works → needs money
3. Hannah doesn't need money
4. So Hannah must enjoy dining halls → not picky → spends time in dining halls

But the model's axiom `H_need_imp : Need Hannah → (¬Picky Hannah ∧ ¬Need Hannah)` is trying to encode the contradiction.

**Should be: DATASET_BUG** (same as case 75 from same story)

## FAITHFUL Cases (8)

V6 says formalization is correct - implies potential GT issue.

In [32]:
faithful_cases = df[df['root_cause_category'] == 'FAITHFUL']['case_idx'].tolist()
print(f'FAITHFUL cases: {faithful_cases}')

FAITHFUL cases: [102, 122, 123, 127, 128, 202, 130, 196]


In [33]:
# Case 202 - Known potential GT error
show_case(202)

CASE 202
Ground Truth: Uncertain
Prediction: True
V6 Category: FAITHFUL
V6 Explanation: The formalization correctly represents the premises, and the conclusion is correctly derived.
Problematic Element: nan

--- PREMISES ---
Ailton Silva, born in 1995, is commonly known as Ailton.
Ailton is a football player who was loaned out to Braga.
Ailton Silva is a Brazilian footballer who plays for Náutico.
Náutico is a football club along with Braga.
Fluminense is a football club.

--- CONCLUSION ---
Ailton Silva was loaned out to a football club.

--- LEAN CODE ---
axiom obj : Type
axiom AiltonSilva : obj
axiom Ailton : obj
axiom Braga : obj
axiom Nautico : obj
axiom Fluminense : obj

axiom FootballClub : obj → Prop
axiom LoanedOutTo : obj → obj → Prop

-- Ailton Silva is commonly known as Ailton (same person)
axiom alias : Ailton = AiltonSilva

-- Ailton was loaned out to Braga
axiom loanAiltonBraga : LoanedOutTo Ailton Braga

-- Náutico and Braga (and Fluminense) are football clubs
axiom fcN

**Case 202 Verdict:** CORRECT - Model's reasoning is valid: Ailton = Ailton Silva, loaned to Braga, Braga is football club => loaned to football club.

In [34]:
# Case 123 - Conservative error (Uncertain -> True)
show_case(123)

CASE 123
Ground Truth: True
Prediction: Uncertain
V6 Category: FAITHFUL
V6 Explanation: The formalization is faithful, and the model's prediction of 'Uncertain' is correct.
Problematic Element: nan

--- PREMISES ---
If something can transport multiple passengers, then they are not one-seaters.
All Tesla Model 3 can transport multiple passengers.
All single-seat electric cars are one-seaters.
All Solo EV cars are single-seat electric cars. 
If John's car is not a Tesla Model 3, then John's car is not a Tesla Model 3 or a single-seat electric car. 

--- CONCLUSION ---
John's car is not a Solo EV.

--- LEAN CODE ---
axiom obj : Type

-- Entities
axiom johns_car : obj

-- Predicates
axiom TransportsMultiplePassengers : obj → Prop
axiom OneSeater : obj → Prop
axiom TeslaModel3 : obj → Prop
axiom SingleSeatElectricCar : obj → Prop
axiom SoloEV : obj → Prop

-- Axioms from the context
axiom A1 : ∀ x : obj, TransportsMultiplePassengers x → ¬ OneSeater x
axiom A2 : ∀ x : obj, TeslaModel3 x → Tr

**Case 123 Analysis (Conservative: Uncertain -> True)**

V6 says: FAITHFUL (formalization correct)

Premise: "If John's car is not a Tesla Model 3, then John's car is not a Tesla Model 3 or a single-seat electric car"

Model's A5: `¬TeslaModel3 → (¬TeslaModel3 ∨ SingleSeatElectricCar)` - **TAUTOLOGY!**

Correct interpretation should be: `¬Tesla → ¬(Tesla ∨ SingleSeat)` = `¬Tesla → (¬Tesla ∧ ¬SingleSeat)`

With correct formalization:
1. ¬Tesla → ¬SingleSeat (from corrected premise)
2. Contrapositive: SingleSeat → Tesla
3. Solo → SingleSeat → OneSeater (from A3, A4)
4. Tesla → TransportsMultiple → ¬OneSeater (from A1, A2)
5. If Solo: OneSeater (from 3) AND Tesla (from 2) → ¬OneSeater (from 4) → CONTRADICTION
6. Therefore: ¬Solo ✓

**Verdict: V6 WRONG** - Should be FORMALIZATION_ERROR, not FAITHFUL

In [35]:
# Show remaining FAITHFUL cases for cross-check
for case_idx in [102, 122, 127, 128, 130, 196]:
    show_case(case_idx)
    print('\n' + '='*70 + '\n')

CASE 102
Ground Truth: False
Prediction: Uncertain
V6 Category: FAITHFUL
V6 Explanation: The formalization accurately represents the premises, and the model's prediction of 'Uncertain' is correct given the information.
Problematic Element: nan

--- PREMISES ---
People who are born in a multiple birth with siblings spend a lot of time hanging out with and playing with their siblings.
If people have siblings who were born together, then they were born in a multiple birth.
If people complain often about annoying siblings, then they have siblings who were born together.
If people live at home, then they do not live with strangers.
If people spend a lot of time hanging out with and playing with their siblings, then they often live at home. 
Luke either is a baby born in a multiple birth and live with strangers, or is not a baby born in a multiple birth and does not live with strangers

--- CONCLUSION ---
Luke complains often about annoying siblings.

--- LEAN CODE ---
axiom obj : Type
axiom

## DATASET_BUG Cases (2)

Premises contain contradictions.

In [36]:
# Case 75 and 159 - Known bad stories (368, 435)
show_case(75)
print('\n')
show_case(159)

CASE 75
Ground Truth: Uncertain
Prediction: True
V6 Category: DATASET_BUG
V6 Explanation: The premises contain a logical contradiction regarding Hannah's need for money.
Problematic Element: Premises about Hannah's need for money

--- PREMISES ---
If people at Mary's school work in student jobs on campus, then they need to earn money to help pay for their college tuition.
If people at Mary's school order takeout frequently in college, then they work in student jobs on campus.
People at Mary's school order takeout frequently in college or enjoy the dining hall meals and recipes.
If people at Mary's school enjoy the dining hall meals and recipes, then they are not picky eaters.
If people at Mary's school enjoy the dining hall meals and recipes, then they spend a lot of their time eating and catching up with friends in the campus dining halls.
Hannah is at Mary's school.
Hannah works in student jobs on campus and if she needs to earn money to help pay for her college tuition, then she is 

**Cases 75, 159 Verdict:** CORRECT - These are from stories 368 and 435 which have contradictory premises.

## Other Cases

In [37]:
# VACUOUS_TRUTH - Case 83
show_case(83)

CASE 83
Ground Truth: False
Prediction: True
V6 Category: VACUOUS_TRUTH
V6 Explanation: The proof derives a contradiction from the faithful representation of the premises, leading to a vacuous truth.
Problematic Element: Theorem statement

--- PREMISES ---
If a restaurant is listed in Yelp’s recommendations, then the restaurant has not received many negative reviews.
All restaurants with a rating greater than four are listed in Yelp’s recommendations.
Some restaurants that do not provide take-out service receive many negative reviews.
All restaurants that are popular among local residents have ratings greater than four.
The Hamden Plaza Subway store has a rating greater than four, or it is popular among local residents.

--- CONCLUSION ---
If the Hamden Plaza Subway store provides take-out service and receives many negative reviews, then its rating is greater than 4 and it does not provide take-out service.

--- LEAN CODE ---
axiom obj : Type
axiom HPS : obj

axiom Listed : obj → Prop


**Case 83 Verdict:** CORRECT - Model proved via vacuous truth (antecedent is impossible).

In [38]:
# GOAL_MISMATCH cases
goal_cases = df[df['root_cause_category'] == 'GOAL_MISMATCH']['case_idx'].tolist()
print(f'GOAL_MISMATCH cases: {goal_cases}')
for case_idx in goal_cases:
    show_case(case_idx)
    print('\n')

GOAL_MISMATCH cases: [100, 40, 114]
CASE 100
Ground Truth: False
Prediction: Uncertain
V6 Category: GOAL_MISMATCH
V6 Explanation: The theorem statement does not match the conclusion provided in the problem.
Problematic Element: theorem statement

--- PREMISES ---
Animals are either invertebrates or vertebrates.
All animals with backbones reproduce by male-and-female mating. 
All vertebrate animals have a backbone.
All bees do not reproduce by male-and-female mating.
All queen bees are bees.
Harry is a bee.

--- CONCLUSION ---
If Harry is either both a vertebrate and an animal with a backbone, or neither a vertebrate nor an animal with a backbone, then Harry is neither an invertebrate nor a queen bee.

--- LEAN CODE ---
axiom obj : Type

axiom Animal : obj → Prop
axiom Invertebrate : obj → Prop
axiom Vertebrate : obj → Prop
axiom Backbone : obj → Prop
axiom ReproMF : obj → Prop
axiom Bee : obj → Prop
axiom QueenBee : obj → Prop

axiom Harry : obj

-- Animals are either invertebrates or 

In [39]:
# REASONING_ERROR - Case 119
show_case(119)

CASE 119
Ground Truth: False
Prediction: Uncertain
V6 Category: REASONING_ERROR
V6 Explanation: The conclusion contradicts the derived result from the premises, indicating a reasoning error.
Problematic Element: Conclusion

--- PREMISES ---
Barutin Cove is a cove named after the Bulgarian settlement of Barutin.
Barutin Cove is on the southwest coast of Snow Island.
Snow Island, Greenwich Island, and Deception Island are located in the South Shetland Islands.
Antarctica is located on the South Shetland Islands. 
If place A is located in place B and place B is located in place C, then place A is located in place C.

--- CONCLUSION ---
Barutin Cove is not located in Antarctica.

--- LEAN CODE ---
axiom obj : Type
axiom BarutinCove : obj
axiom SnowIsland : obj
axiom GreenwichIsland : obj
axiom DeceptionIsland : obj
axiom SouthShetlandIslands : obj
axiom Antarctica : obj

axiom LocatedIn : obj → obj → Prop

-- Facts from the context
axiom A1 : LocatedIn BarutinCove SnowIsland
axiom A2 : Loc

In [40]:
# MISSING_PREMISE - Case 99
show_case(99)

CASE 99
Ground Truth: True
Prediction: Uncertain
V6 Category: MISSING_PREMISE
V6 Explanation: The premise that bees are animals is missing, preventing the application of axioms about animals to Harry.
Problematic Element: Missing premise that bees are animals

--- PREMISES ---
Animals are either invertebrates or vertebrates.
All animals with backbones reproduce by male-and-female mating. 
All vertebrate animals have a backbone.
All bees do not reproduce by male-and-female mating.
All queen bees are bees.
Harry is a bee.

--- CONCLUSION ---
Harry is an invertebrate or a queen bee.

--- LEAN CODE ---
axiom obj : Type
axiom Animal : obj → Prop
axiom Invertebrate : obj → Prop
axiom Vertebrate : obj → Prop
axiom Backbone : obj → Prop
axiom MatesMF : obj → Prop
axiom Bee : obj → Prop
axiom QueenBee : obj → Prop
axiom Harry : obj

-- Animals are either invertebrates or vertebrates.
axiom A1 : ∀ x : obj, Animal x → (Invertebrate x ∨ Vertebrate x)

-- All vertebrate animals have a backbone.
axi

**Case 99 Verdict:** DEBATABLE - NL_AMBIGUITY / WORLD_KNOWLEDGE

V6 says MISSING_PREMISE: "The premise that bees are animals is missing"

Premise says: "Harry is a bee" and other premises talk about "animals".

The model assumed bees are animals (world knowledge). 
- This is technically correct - bees ARE animals
- But the premise doesn't explicitly state this

**Could be classified as:**
- MISSING_PREMISE (V6's choice) - if we require explicit statements
- WORLD_KNOWLEDGE - if we allow reasonable inferences
- NL_AMBIGUITY - the dataset might expect common sense

## Summary of V6 Classification Errors

### Verified Wrong Classifications

| Case | V6 Says | Issue | Should Be |
|------|---------|-------|-----------|
| **29** | AXIOM_FABRICATION | V6 self-contradicts: explanation says "all axioms faithful" | FAITHFUL |
| **36** | AXIOM_FABRICATION | Wrong implication direction, not fabrication | FORMALIZATION_ERROR |
| **46** | AXIOM_FABRICATION | "Student" IS in premise, model interpreted differently | NL_AMBIGUITY |
| **77** | AXIOM_FABRICATION | Same story 368 as case 75 (DATASET_BUG) | DATASET_BUG |
| **89** | AXIOM_FABRICATION | Fabricated axiom exists but isn't used in proof | FAITHFUL |
| **92** | AXIOM_FABRICATION | Premise literally says "A deposit is required..." | FAITHFUL |
| **118** | AXIOM_FABRICATION | "Bulgarian settlement" → in Bulgaria is reasonable | WORLD_KNOWLEDGE |
| **123** | FAITHFUL | Tautological axiom due to NL ambiguity | FORMALIZATION_ERROR |

### V6 Accuracy by Category

| Category | Total | Correct | Wrong | Accuracy |
|----------|-------|---------|-------|----------|
| AXIOM_FABRICATION | 11 | ~4 | ~7 | ~36% |
| FAITHFUL | 8 | ~2 | ~6 | ~25% |
| DATASET_BUG | 2 | 2 | 0 | 100% |
| VACUOUS_TRUTH | 1 | 1 | 0 | 100% |
| GOAL_MISMATCH | 3 | ? | ? | ? |
| REASONING_ERROR | 1 | ? | ? | ? |
| MISSING_PREMISE | 1 | ~0.5 | ~0.5 | 50% |

**Overall V6 Accuracy: ~40-50%** (roughly half the classifications are wrong or debatable)

### Key Problems with V6

1. **Self-contradiction**: Case 29 - category says FABRICATION, explanation says FAITHFUL

2. **Hallucination**: Case 92 - V6 claims axiom isn't in premises, but it literally is

3. **Inconsistency across same story**: Cases 75 vs 77 - both from story 368, different classifications

4. **AXIOM_FABRICATION is too broad** - conflates:
   - True fabrication (making up facts not inferable)
   - NL ambiguity (different valid interpretations)
   - World knowledge (bees are animals, Bulgarian settlements are in Bulgaria)
   - Formalization errors (wrong direction)

5. **FAITHFUL misused** - doesn't catch:
   - Tautological axioms (Case 123)
   - Unused axioms that don't affect reasoning

6. **Doesn't check axiom usage**: Case 89 has unused fabricated axiom but proof is valid

### Proposed Better Classification Taxonomy

| Error Type | Description | Examples |
|------------|-------------|----------|
| PURE_FABRICATION | Made up fact not derivable from premises | Case 70 (KO is a stock) |
| WORLD_KNOWLEDGE | Used common sense not in premises | Case 118 (Bulgarian → in Bulgaria) |
| NL_AMBIGUITY | Multiple valid interpretations of NL | Case 46 (students), Case 123 (or/nor) |
| WRONG_DIRECTION | Reversed implication direction | Case 36 (conductors lead orchestras) |
| TAUTOLOGY | Axiom is trivially true | Case 123 (¬A → ¬A ∨ B) |
| SCOPE_ERROR | Quantifier scope wrong | - |
| DATASET_BUG | Contradictory premises | Cases 75, 77, 159 |
| WRONG_GT | Ground truth is questionable | Cases 89, 202 |
| GOAL_MISMATCH | Theorem doesn't match conclusion | Cases 40, 100, 114 |
| FAITHFUL | Formalization is truly correct | Case 202 |